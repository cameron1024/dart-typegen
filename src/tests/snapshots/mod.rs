// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
