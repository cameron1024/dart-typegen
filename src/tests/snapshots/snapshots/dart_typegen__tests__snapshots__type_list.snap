---
source: src/tests/snapshots/mod.rs
expression: dart
---
final class Foo {
  final List<String> x;
  final List<int> y;

  const Foo({required this.x, required this.y});

  static FooBuilder builder({required List<String> x, required List<int> y}) =>
      FooBuilder(x: x, y: y);
  FooBuilder toBuilder() => FooBuilder(x: x, y: y);

  Map<String, dynamic> toJson() => {"x": x, "y": y};
  factory Foo.fromJson(Map<String, dynamic> json) =>
      Foo(x: json["x"] as List<String>, y: json["y"] as List<int>);

  @override
  String toString() =>
      "Foo("
      "x: $x, "
      "y: $y"
      ")";
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) {
      return true;
    }
    if (other is! Foo) {
      return false;
    }
    if (x.length != other.x.length) return false;
    for (var i = 0; i < x.length; i++) {
      if (x[i] != other.x[i]) return false;
    }
    if (y.length != other.y.length) return false;
    for (var i = 0; i < y.length; i++) {
      if (y[i] != other.y[i]) return false;
    }
    return true;
  }

  @override
  int get hashCode => Object.hashAll([x, y]);
}

/// Builder class for [Foo]
final class FooBuilder {
  List<String> x;
  List<int> y;

  FooBuilder({required this.x, required this.y});

  Foo build() => Foo(x: x, y: y);
}
